data/src/main/java/org/akvo/flow/data/datasource/preferences/SharedPreferencesDataSource.java

```diff
@@ -187,11 +163,75 @@ private void setInt(String key, int value) {
         return Observable.just(true);
     }
 
+    public Observable<Boolean> mobileUploadSet() {
+        return Observable.just(preferences.contains(KEY_CELL_UPLOAD));
+    }
+
+    public Observable<Boolean> saveApkData(ApkData apkData) {
+        setString(KEY_APK_DATA, gsonMapper.write(apkData, ApkData.class));
+        return Observable.just(true);
+    }
+
+    public Observable<Boolean> clearAppUpdateNotified() {
+        removePreference(KEY_APP_UPDATE_LAST_NOTIFIED);
+        return Observable.just(true);
+    }
+
+    @Nullable
+    public Observable<ApkData> getApkData() {
+        String apkDataString = preferences.getString(KEY_APK_DATA, null);
+        if (apkDataString == null) {
+            return Observable.just(ApkData.NOT_SET_VALUE);
+        }
+        return Observable.just(gsonMapper.read(apkDataString, ApkData.class));
+    }
+
+    public Observable<Boolean> saveAppUpdateNotifiedTime() {
+        setLong(KEY_APP_UPDATE_LAST_NOTIFIED, System.currentTimeMillis());
+        return Observable.just(true);
+    }
+
+    public Observable<Long> getAppUpdateNotifiedTime() {
+        return Observable.just(getLong(KEY_APP_UPDATE_LAST_NOTIFIED, LONG_VALUE_UNSET));
+    }
+
+    protected String getString(String key, String defaultValue) {
+        return preferences.getString(key, defaultValue);
+    }
+
+    protected void setString(String key, String value) {
+        preferences.edit().putString(key, value).apply();
+    }
+
+    private boolean getBoolean(String key, boolean defValue) {
```

Comment:
its a small comment but may be help to keep a consistent format here.. i.e. either getters first and then setters, or getter/setter pairs together.